artemis/cleanup.py

```diff
@@ -67,9 +71,26 @@ def process(self, *args, **kwargs):  # type: ignore
         logger.info("Queue for %s is cleaned up", old_module)
 
 
+def _cleanup_scheduled_tasks() -> None:
+    karton_state = KartonState(backend=KartonBackend(config=KartonConfig()))
+    finished_analysis_ids = []
+    for analysis in db.list_analysis():
+        if analysis["id"] not in karton_state.analyses or len(karton_state.analyses[analysis["id"]].pending_tasks) == 0:
+            finished_analysis_ids.append(analysis["id"])
+
```

Comment:
add an assert we have also **unfinished** analyses ids, in case we made a mistake in lines 75-79 (e.g. the key format changed) ;)

Reply:
I've added additionaly a flag to control that behaviour, this way e.g: running it locally will not throw errors in case of zero activity.